subcell_pipeline/simulation/readdy/loader.py

```diff
@@ -1,83 +1,96 @@
-#!/usr/bin/env python
+"""Class for loading and shaping ReaDDy trajectories."""
 
-import os
-from typing import Any, List, Optional
+from typing import Any, Optional
 
 import numpy as np
 import readdy
+from io_collection.keys.check_key import check_key
+from io_collection.load.load_pickle import load_pickle
+from io_collection.save.save_pickle import save_pickle
 from tqdm import tqdm
 
-from .readdy_data import FrameData, ParticleData, TopologyData
+from .data_structures import FrameData, ParticleData, TopologyData
 
 
 class ReaddyLoader:
+    """
+    Load and shape data from a ReaDDy trajectory.
+
+    Trajectory is loaded from the simulation output h5 file of the .dat pickle
+    file. If a .dat pickle location and key are provided, the loaded trajectory
+    is saved to the given location for faster reloads.
+    """
+
+    _readdy_trajectory: Optional[readdy.Trajectory]
+    """ReaDDy trajectory object."""
+
+    _trajectory: Optional[list[FrameData]]
+    """List of FrameData for trajectory."""
+
+    h5_file_path: str
+    """Path to the ReaDDy .h5 file or .dat pickle file."""
+
+    min_time_ix: int
+    """First time index to include."""
+
+    max_time_ix: int
+    """Last time index to include."""
+
+    time_inc: int
+    """Include every time_inc timestep."""
+
+    timestep: float
+    """Real time for each simulation timestep."""
+
+    pickle_location: Optional[str]
+    """Location to save pickle file (AWS S3 bucket or local path)."""
+
+    pickle_key: Optional[str]
+    """Name of pickle file (AWS S3 bucket or local path)."""
+
     def __init__(
         self,
         h5_file_path: str,
         min_time_ix: int = 0,
         max_time_ix: int = -1,
         time_inc: int = 1,
         timestep: float = 100.0,
-        save_pickle_file: bool = False,
+        pickle_location: Optional[str] = None,
```

Comment:
Does this function need documentation or the class attribute descriptions are sufficient?

Reply:
since it's the constructor I usually just rely on the class documentation